code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm

```diff
@@ -0,0 +1,529 @@
+			vent.update_appearance(UPDATE_ICON)
+		if ("scrubbing")
+			if (isnull(scrubber))
+				return TRUE
+
+			scrubber.set_scrubbing(!!params["val"], user)
+		if ("widenet")
+			if (isnull(scrubber))
+				return TRUE
+
+			scrubber.set_widenet(!!params["val"])
+		if ("toggle_filter")
+			if (isnull(scrubber))
+				return TRUE
+
+			scrubber.toggle_filters(params["val"])
+		if ("mode")
+			select_mode(user, text2path(params["mode"]))
+			investigate_log("was turned to [selected_mode.name] mode by [key_name(user)]", INVESTIGATE_ATMOS)
+
+		if ("set_threshold")
+			var/threshold = text2path(params["threshold"]) || params["threshold"]
+			var/datum/tlv/tlv = tlv_collection[threshold]
+			if(isnull(tlv))
+				return
+			var/threshold_type = params["threshold_type"]
+			var/value = params["value"]
+			tlv.set_value(threshold_type, value)
+			investigate_log("threshold value for [threshold]:[threshold_type] was set to [value] by [key_name(usr)]", INVESTIGATE_ATMOS)
+
+			var/turf/our_turf = get_turf(src)
+			var/datum/gas_mixture/environment = our_turf.return_air()
+			check_air_dangerlevel(our_turf, environment, environment.temperature)
+
+		if("reset_threshold")
+			var/threshold = text2path(params["threshold"]) || params["threshold"]
+			var/datum/tlv/tlv = tlv_collection[threshold]
+			if(isnull(tlv))
+				return
+			var/threshold_type = params["threshold_type"]
+			tlv.reset_value(threshold_type)
+			investigate_log("threshold value for [threshold]:[threshold_type] was reset by [key_name(usr)]", INVESTIGATE_ATMOS)
+
+			var/turf/our_turf = get_turf(src)
+			var/datum/gas_mixture/environment = our_turf.return_air()
+			check_air_dangerlevel(our_turf, environment, environment.temperature)
+
+		if ("alarm")
+			if (alarm_manager.send_alarm(ALARM_ATMOS))
+				danger_level = AIR_ALARM_ALERT_SEVERE
+
+		if ("reset")
+			if (alarm_manager.clear_alarm(ALARM_ATMOS))
+				danger_level = AIR_ALARM_ALERT_NONE
+
+	update_appearance()
+
+	return TRUE
+
+/obj/machinery/airalarm/update_appearance(updates)
+	. = ..()
+
+	if(panel_open || (machine_stat & (NOPOWER|BROKEN)) || shorted)
+		set_light(0)
+		return
+
+	var/area/our_area = get_area(src)
+	var/color
+	switch(max(danger_level, !!our_area.active_alarms[ALARM_ATMOS]))
+		if(0)
+			color = "#03A728" // green
+		if(1)
+			color = "#EC8B2F" // yellow
+		if(2)
+			color = "#DA0205" // red
+
+	set_light(1.4, 1, color)
+
+/obj/machinery/airalarm/update_icon_state()
+	if(panel_open)
+		switch(buildstage)
+			if(AIRALARM_BUILD_COMPLETE)
+				icon_state = "alarmx"
+			if(AIRALARM_BUILD_NO_WIRES)
+				icon_state = "alarm_b2"
+			if(AIRALARM_BUILD_NO_CIRCUIT)
+				icon_state = "alarm_b1"
+		return ..()
+
+	icon_state = "alarmp"
+	return ..()
+
+/obj/machinery/airalarm/update_overlays()
+	. = ..()
+
+	if(panel_open || (machine_stat & (NOPOWER|BROKEN)) || shorted)
+		return
+
+	var/area/our_area = get_area(src)
+	var/state
+	switch(max(danger_level, !!our_area.active_alarms[ALARM_ATMOS]))
```

Comment:
think we should just make it a proc? oh but that'd do almost nothing. guess it would be useful to pull out just to name it, and for reuse in future

Reply:
Ill try and detach danger_level from area code and keep it local. This is kinda stupid.